File: src/main/java/org/example/shapecarvejavafx/Output.java
package org.example.shapecarvejavafx;

import java.util.Arrays;
import java.util.List;
import java.util.Objects;

public final class Output {
    private final List<Integer> volume;
    private final List<Integer> dims;

    Output(int[] volume, int[] dims) {
        this.volume = Arrays.stream(volume).boxed().toList();
        this.dims = Arrays.stream(dims).boxed().toList();
    }

    public List<Integer> volume() {
        return volume;
    }

    public List<Integer> dims() {
        return dims;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == this) return true;
        if (obj == null || obj.getClass() != this.getClass()) return false;
        var that = (Output) obj;
        return this.volume.equals(that.volume) &&
                this.dims.equals(that.dims);
    }

    @Override
    public int hashCode() {
        return Objects.hash(volume.hashCode(), dims.hashCode());
    }

    @Override
    public String toString() {
        return "output[" +
                "volume=" + volume.toString() + ", " +
                "dims=" + dims.toString() + ']';
    }
}

File: src/main/java/org/example/shapecarvejavafx/Output.kt
package org.example.shapecarvejavafx

import java.util.*

class Output internal constructor(volume: IntArray?, dims: IntArray?) {
    private val volume: List<Int> = Arrays.stream(volume).boxed().toList()
    private val dims: List<Int> = Arrays.stream(dims).boxed().toList()

    fun volume(): List<Int> {
        return volume
    }

    fun dims(): List<Int> {
        return dims
    }

    override fun equals(obj: Any?): Boolean {
        if (obj === this) return true
        if (obj == null || obj.javaClass != this.javaClass) return false
        val that = obj as Output
        return this.volume == that.volume && (this.dims == that.dims)
    }

    override fun hashCode(): Int {
        return Objects.hash(volume.hashCode(), dims.hashCode())
    }

    override fun toString(): String {
        return "output[" +
                "volume=" + volume.toString() + ", " +
                "dims=" + dims.toString() + ']'
    }
}
